src/cursor.ts

```diff
@@ -194,10 +221,40 @@ export async function executeWithCursorPagination<
         const comparison = field.direction === defaultDirection ? ">" : "<";
         const value = cursor[field.key as keyof typeof cursor];
 
-        const conditions = [eb(field.expression, comparison, value)];
+        let conditions = [eb(field.expression, comparison, value)];
+        if (field.nullable && field.dataType) {
+          const boundaryValue = getBoundaryValue(
+            field.direction,
+            field.dataType
+          );
+          if (reversed) {
+            conditions = [
+              eb(
+                field.expression,
+                comparison,
+                fn.coalesce(
+                  sql.val(value),
+                  cast(sql.val(boundaryValue), field.dataType)
```

Comment:
The `cast` is a little bit of an oddity here. The compiled query runs fine in Postgres without casting. But somewhere in `node-pg` it fails without this cast.

Reply:
Also, I had to update the `kysely` dependency since `cast` was not available in the version thats on `main`